google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/facet/AppEngineStandardFacetEditor.java

```diff
@@ -166,50 +167,42 @@ private boolean isManagedLibrariesEnabled() {
 
     @Override
     public void afterLibraryAdded(final Library addedLibrary) {
-      AppEngineStandardMavenLibrary library = AppEngineStandardMavenLibrary
-          .getLibraryByMavenDisplayName(addedLibrary.getName());
-
-      if (library == null) {
-        return;
-      }
-
-      final DependencyScope scope = library.getScope();
-
-      new WriteAction() {
-        @Override
-        protected void run(@NotNull Result result) throws Throwable {
-          ModuleRootManager manager = ModuleRootManager.getInstance(context.getModule());
-          ModifiableRootModel model = manager.getModifiableModel();
-
-          model.addLibraryEntry(addedLibrary);
-
-          for (OrderEntry orderEntry : model.getOrderEntries()) {
-            if (orderEntry.getPresentableName().equals(addedLibrary.getName())) {
-              ((ExportableOrderEntry) orderEntry).setScope(scope);
-            }
+      AppEngineStandardMavenLibrary.getLibraryByMavenDisplayName(addedLibrary.getName()).ifPresent(
```

Comment:
I think factoring out the lambda contents into a new method would actually decrease readability, since that lambda essentially contains the bulk of logic of the method. i.e., `afterLibraryAdded()` would be a one-liner with a method reference, which voids the utility of the reference method.

We could get rid of the ifPresent() by making sure that getLibraryByMavenDisplayName() can always return something, though I'm not familiar enough with the code to say if that can be done.

Finally, I think the following is actually pretty readable, but feel free to disagree.

![screenshot from 2017-02-22 15 25 46](https://cloud.githubusercontent.com/assets/3290856/23230992/02623ef8-f914-11e6-9147-33892c25ab2b.png)